Rideable Carps - Several types of Lassos

DrDuckedGoose opened the pull request:
## About The Pull Request

This is not an exact port! This is an interpretation of someone's suggestion, no external code was referenced, or needs attributing.

Introduces several types of lassos.
- Basic
- Primal
- Antag
- Debug
These lassos can be used to tame animals. Specifically, the basic lasso can tame space-carp & cows, this type can be crafted. The primal lasso can also be crafted, exclusively by ash-walkers, can be used to tame certain lavaland fauna. The antag lasso can tame anything outside of a specified blacklist, bubblegum & other bosses, can be bought from any uplink for 3TC. The debug lasso tames anything.
Tamed animals can be rode & given commands through speech and radial menus(alt-click).
Radial menu icons are procedurally generated, based on the animal's icon.

## Why It's Good For The Game

Unique item, free content.

## Testing Photographs and Procedure
<details>
<summary>Screenshots&Videos</summary>
Coder sprites - Subject to change

![image](https://user-images.githubusercontent.com/40559528/177014313-a9308660-eb28-41a0-94ba-f8d727f81b5d.png)
![image](https://user-images.githubusercontent.com/40559528/177014183-f45b77bf-3c6f-41d1-9765-f7b647255933.png)
![image](https://user-images.githubusercontent.com/40559528/177014185-643bb451-3be5-466d-b9c7-8d82804ea25b.png)
![image](https://user-images.githubusercontent.com/40559528/177014192-16c60d1e-e02f-4951-ba52-7776c6b751dc.png)
![image](https://user-images.githubusercontent.com/40559528/177014197-7112c533-466c-46c9-a0e5-2f002c3ae035.png)
![image](https://user-images.githubusercontent.com/40559528/177014198-d03d8dca-14e9-4088-b331-bb0bafed8813.png)
![image](https://user-images.githubusercontent.com/40559528/177014990-272ee6c8-b4e9-40f6-93ef-b805691c1bac.png)
![image](https://user-images.githubusercontent.com/40559528/177030322-046b2bb2-48f3-490d-b37f-27a4ec08e8d2.png)
![image](https://user-images.githubusercontent.com/40559528/177950072-39f56fe4-9bf2-404c-b3ab-800dcdb1ead4.png)
![image](https://user-images.githubusercontent.com/40559528/177950171-02c9c5c5-c7fe-4ec3-b6f9-325e7ee8dced.png)
![image](https://user-images.githubusercontent.com/40559528/177950277-2090907c-830a-422b-8250-5a193c69359a.png)
![image](https://user-images.githubusercontent.com/40559528/177950495-a64f09ce-4055-4612-a0eb-cd36f502751d.png)


https://user-images.githubusercontent.com/40559528/177044050-c7368168-31cf-4863-aaca-fd26e0650a47.mp4



</details>

## Changelog
:cl:
add: Adds lasso items
add: Adds tamed riding component subtype, toggles AI & sometimes spacewalking
add: Adds antag lasso to syndicate uplink
add: Add new AI for tamed animals
add: Add tamed component
/:cl:

![image](https://user-images.githubusercontent.com/40559528/177014267-9a9fa90b-5d1e-4e4d-afc5-b13ff9ef07d8.png)

DrDuckedGoose committed:
Merge https://github.com/BeeStation/BeeStation-Hornet into annoying-thing Merge conflicts :)

DrDuckedGoose committed:
initial

DrDuckedGoose committed:
Merge https://github.com/BeeStation/BeeStation-Hornet into carp-ride

Hee Hoo merge conflicts

Penwin0 commented on code/game/objects/items/carp_lasso.dm:
```diff
@@ -0,0 +1,44 @@
+	///Ref to timer
+	var/timer
+	///Ref to lasso'd carp
+	var/mob/living/simple_animal/hostile/carp/carp_target
```
Make sure this doesn't cause harddels

DrDuckedGoose replied:
No worries, smokey
![image](https://user-images.githubusercontent.com/40559528/177014493-4bb7829a-c948-400a-aa8d-8fdd0c4b5b3a.png)

DrDuckedGoose committed:
Handle Hard Del & Implement beams

- Dedicate function and signal to handle hard deleting
- Lasso now creates a beam when used, imitating rope.

DrDuckedGoose committed:
Implement Carp Command Component - 3 7 22

- Allow carps to be given commands and interact with crew

DrDuckedGoose committed:
Further Implementation - 3 7 22

DrDuckedGoose committed:
Fix Signals - 3 7 22

DrDuckedGoose committed:
Merge https://github.com/BeeStation/BeeStation-Hornet into carp-ride
hee hoo

DrDuckedGoose committed:
Fix Voice Commands & Ally system - 3 7 22

- Fix voice commands being funky
- Fix ally system being used irrespectively
- Add better interpretation to voice commands & commands in general

DrDuckedGoose committed:
Conga Carp - 4 7 22

- Handle some hard dels
- Carps try not to get in each other's way

Dejaku51 reviewed (commented):
could you change it so it use the AI controller like the dog AI?

DrDuckedGoose committed:
Anguish - 6 7 22

- Rewrites taming / riding system
- Changes component to rely on AI controller
- Item can now be abused by admins to tame almost anything
- Component is now general and not dedicate to carps.
- Icons now use base resources and generate from there.
- Adds subtype that can tame anything

DrDuckedGoose committed:
Fixes - 6 7 22

DrDuckedGoose committed:
Crafting Recipes - 7 7 22

- Basic lassos can be crafted with leather
- Introduces primal lassos, can tame certain lavaland fauna
    - primal lassos can be crafted by ash walkers
- Add more taming option to basic lasso
- Add less taming options to antag lasso
- Lowers cost from 5TC to 3TC

DrDuckedGoose committed:
Update uplink_items.dm

DrDuckedGoose committed:
Spelling - 8 7 22

PowerfulBacon reviewed (approved):
cool
